File: lib/result/ReceiveData.java
package com.teamlab.communication.wireless.lpr9201.result;

/**
 * リザルト ReceiveData
 */
public class ReceiveData extends Result {
	/**
	 * 
	 */
	public static final int RESULT_CODE = 0x83;

	/**
	 * 
	 */
	public static final int DATA_LENGTH_SIZE = 2; // byte

	/**
	 * 
	 * @param datas
	 */
	public ReceiveData(int[] datas) {
		super(RESULT_CODE, DATA_LENGTH_SIZE, datas);
	}
}

File: lib/result/receive_data.js
/* jslint node: true */
"use strict"

/**
 * リザルト ReceiveData
 */

exports.ReceiveData = ReceiveData;

/**
 *  結果コード
 */
ReceiveData.RESULT_CODE = 0x83;

/**
 * データ長バイトサイズ
 */
ReceiveData.DATA_LENGTH_BYTE_SIZE = 2; // byte

/**
 * @param {Result} result
 */
function ReceiveData(result) {
  this.result = result;
  this.datas = result.datas;
}

/**
 * パース可能か
 * @param {Result} result
 */
ReceiveData.canParse = function(result) {
  return result !== null && result.resultCode === ReceiveData.RESULT_CODE;
}
